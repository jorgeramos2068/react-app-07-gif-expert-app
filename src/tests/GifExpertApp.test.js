import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Tests for <GifExpertApp />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<GifExpertApp />);
  });

  test('Should display <GifExpertApp /> correctly', () => {
    wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show a list of categories', () => {
    const categories = ['Batman', 'Robin'];
    wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);

  });
});
