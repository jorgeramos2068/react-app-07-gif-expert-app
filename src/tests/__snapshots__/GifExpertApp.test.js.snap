// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <GifExpertApp /> Should display <GifExpertApp /> correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests for <GifExpertApp /> Should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Batman"
      key="Batman"
    />
    <GifGrid
      category="Robin"
      key="Robin"
    />
  </ol>
</Fragment>
`;
